Postgres metrics

rishi-bilv opened the pull request.

rishi-bilv committed:
feat: integrate PostgreSQL user service and update authentication routes

- Introduced a new PostgresUserService for handling user operations with PostgreSQL.
- Updated authentication routes to support PostgreSQL-based registration and login.
- Modified the AuthService to utilize the new PostgresUserService for user management.
- Enhanced user validation to ensure either username or email is provided during registration.
- Added PostgreSQL connection settings to the configuration file.
- Updated requirements to include necessary PostgreSQL libraries.

rishi-bilv committed:
feat: integrate PostgreSQL metrics and assistant usage tracking

- Added PostgreSQL support for logging review metrics and assistant usage.
- Implemented new classes for managing review metrics and assistant usage in PostgreSQL.
- Enhanced existing review endpoints to log metrics in PostgreSQL after code reviews.
- Introduced methods for retrieving metrics from PostgreSQL, including issue severity distribution and usage statistics.
- Added a connection test script for PostgreSQL to ensure connectivity and troubleshoot issues.
- Updated the application to initialize the PostgreSQL schema with new metric tables.

saikrishnakompelly-bilvantis commented on app.py:
```diff
@@ -230,6 +234,54 @@ async def read_users_me(current_user: dict = Depends(auth_service.get_current_us
+    )
+    return {"message": "[PG] User registered successfully", "user_id": user_id}
+
+@app.post(f"{BASE_URL}/auth/pg/login", tags=["Authentication"], summary="PG Login (dummy)")
```
remove pg prefix for all methods once mongo is removed since we only need the existing mnethods without needing any changes on the client side

saikrishnakompelly-bilvantis commented on src/services/user/user_service_postgres.py:
rename this file to userservice once the mongo services aare removed

saikrishnakompelly-bilvantis commented on src/metrics/pg_generate.py:
rename file to original service name

saikrishnakompelly-bilvantis commented on src/metrics/pg_retrieve.py:
```diff
@@ -0,0 +1,303 @@
+    return dt.strftime("%Y-%m")
+
+
+class PgRetrieveMetrics:
```
Cklass names should be same as original

saikrishnakompelly-bilvantis commented on src/services/user/test_psycopg2_connection.py:
connection logs to be added at application startup

saikrishnakompelly-bilvantis reviewed (changes requested):
resolve all the comments added

rishi-bilv committed:
refactor: transition from MongoDB to PostgreSQL for user and metrics services

- Removed MongoDB dependencies for user and metrics services, transitioning to PostgreSQL exclusively.
- Updated UserService and AuthService to utilize PostgreSQL for user management and authentication.
- Refactored metrics logging to use PostgreSQL, including new classes for managing review metrics and assistant usage.
- Adjusted application initialization to set up PostgreSQL schema and connections.
- Cleaned up unused MongoDB-related code and imports, streamlining the application structure.

rishi-bilv committed:
refactor: standardize PostgreSQL connection settings and enhance database setup

- Updated PostgreSQL connection settings in user service and configuration files to use consistent naming conventions.
- Introduced a new function to load PostgreSQL configuration with sensible defaults and fallbacks.
- Enhanced the database setup process to create necessary tables if they do not exist, ensuring a smoother initialization experience.
- Refactored metrics retrieval code to improve type annotations and clarity in filter handling.

saikrishnakompelly-bilvantis committed:
Merge branch 'hsbc-develop' into postgres-metrics

saikrishnakompelly-bilvantis reviewed (approved):
Done
